### PR TITLE
Do not set capabilities for Elasticsearch < 8.x

### PR DESCRIPTION
**Pull request by naemono**
Elasticsearch v7 does not work well when `securityContext.Capabilities` is set to `Drop: "all"`. Both the initContainer that prepares the filesystem fails when attempting to preserve ownership, and the main Elasticsearch container fails with `chroot: cannot change root directory to '/': Operation not permitted` as the containers likely run as the `root` user pre v8.

### CODE REVIEW

File: pkg/controller/elasticsearch/securitycontext/securitycontext.go [hunk 1]
```diff
@@ -12,22 +12,30 @@ import (
...
-		Capabilities: &corev1.Capabilities{
-			Drop: []corev1.Capability{"ALL"},
-		},
 		Privileged:               ptr.Bool(false),
 		ReadOnlyRootFilesystem:   ptr.Bool(enableReadOnlyRootFilesystem),
 		AllowPrivilegeEscalation: ptr.Bool(false),
 	}
-	if ver.LT(MinStackVersion) {
+	if ver.LT(DropCapabilitiesMinStackVersion) {
+		return sc
```
Comment (barkbay) [inline pkg/controller/elasticsearch/securitycontext/securitycontext.go:33]: Maybe add a unit test to check that `Capabilities` are not set if < 8.0.0

**Comment by naemono** [comment]
buildkite test this -f p=gke -m s=8.7.0,s=7.17.10

**Review by barkbay — APPROVED** [review]
LGTM

File: pkg/controller/elasticsearch/securitycontext/securitycontext.go (hunk 1 above)
Comment (naemono) [reply pkg/controller/elasticsearch/securitycontext/securitycontext.go:33]: > Maybe add a unit test to check that `Capabilities` are not set if < 8.0.0

Will do. Gotta fix some other tests as well...

**Pull request merged**